feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,118 @@
+
+        _check_param_drop_original(drop_original=drop_original)
+        self.drop_original = drop_original
+
+        self.variables = None
+
+    @staticmethod
+    def _check_column_name(column_name: str) -> str:
+        if not isinstance(column_name, str):
+            raise ValueError(
+                "column_name takes only string as value. "
+                f"Got {column_name} instead."
+            )
+
+        return column_name
+
+    def transform(self, X: pd.DataFrame):
+        """
+        Compute the distance between the two coordinates given using the Haversine formula
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shame = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_new: Pandas dataframe.
+            The original dataframe plus the distance between the given coordinates.
+        """
+        X = self._check_transform_input_and_state(X)
+        X = self._check_lat_lon_columns_are_in_df(X)
+        X = self._check_correctness_of_coordinates(X)
+
+        self.compute_distance(X)
+
+        if self.drop_original:
+            X.drop(
+                columns=[
+                    self.a_latitude,
+                    self.a_longitude,
+                    self.b_latitude,
+                    self.b_longitude,
+                ],
+                inplace=True)
+
+        return X
+
+    def compute_distance(self, X: pd.DataFrame):
+        # convert latitude and longitude in radians
+        phi_1 = np.radians(X[self.a_latitude])
+        phi_2 = np.radians(X[self.b_latitude])
+        lambda_1 = np.radians(X[self.a_longitude])
+        lambda_2 = np.radians(X[self.b_longitude])
+
+        # compute delta, i.e., difference, between radians
+        delta_phi = phi_2 - phi_1
+        delta_lambda = lambda_2 - lambda_1
+
+        # compute distance using Haversine formula
+        inner_part = np.sin(delta_phi / 2) ** 2 + np.cos(phi_1) * np.cos(phi_2) * np.sin(delta_lambda / 2) ** 2
+        X[self.output_column_name] = self.EARTH_RADIUS * 2 * np.arcsin(np.sqrt(inner_part))
+
+    def _check_lat_lon_columns_are_in_df(self, X) -> pd.DataFrame:
+        df_columns = set(X.columns)
+        input_columns = {self.a_latitude, self.a_longitude, self.b_latitude, self.b_latitude}
+
+        if input_columns.issubset(df_columns) is False:
+            raise ValueError(f'The columns {input_columns.difference(df_columns)} were not found in the dataframe.')
+
+        return X
+
+    def _check_correctness_of_coordinates(self, X: pd.DataFrame) -> pd.DataFrame:
+        irregular_latitudes = X[(X[self.a_latitude].abs() > 90) | (X[self.b_latitude].abs() > 90)]
+        irregular_longitudes = X[(X[self.a_longitude].abs() > 180) | (X[self.b_longitude].abs() > 180)]
+
+        if irregular_latitudes.empty is False:
+            raise ValueError(f'The dataframe contains irregular latitudes: {irregular_latitudes}')
```

Comment:
what does this error message return? the portion of the df with irregular latitudes?  I think it would be more useful to the user if the error says "variables so and so have latitude values greater than 90 for example.

Also, do we want to raise an error? or we want to return nan? or we want to make it optional? for example control the output through a parameter that I think we call error in other transformers, if ignore, we set to nan, if raise we raise these errors you have here?

Reply:
Yes, I see your point.

For me the responsibility should be on the final user, i.e., if you call the transformer on a column with coordinates, then you are sure that the coordinates contained in the column are correct. If you forgot to check that, we will remember you raising an error. 

But I'm open to also do differently :-)

Reply:
Let's go with your suggestion then.